### PR TITLE
Use redux in component

### PR DESCRIPTION
**Pull request by Yacoubou-seidou**
### Use redux in component
- Wrap the App with the `Provider` component from `react-redux`
- Pass store into the `Provider`
- Create initialState variable for the bookstore with data
- Display books from the slice
- Dispatch action using `useDispatch`
- Add `add` book `Button` and event handler with attribute ( id, title, author)
- Add `remove` book `Button` and event handler with (id)

### CODE REVIEW

File: src/components/ShowBook.js [hunk 1]
```diff
@@ -1,12 +1,13 @@
 import React from 'react';
+// eslint-disable-next-line import/no-extraneous-dependencies
```
Comment (TedLivist) [inline src/components/ShowBook.js]: - [ ] Kindly avoid disabling linters as they help us write clean code. Remove this disabling line and fix the error.

File: src/components/AddForm.js [hunk 2]
```diff
@@ -1,11 +1,37 @@
-import React from 'react';
+import { useState, React } from 'react';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { useDispatch } from 'react-redux';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { v4 as uuidv4 } from 'uuid';
+// eslint-disable-next-line import/extensions
+import { addBook } from '../redux/books/booksSlice';
```
Comment (TedLivist) [inline src/components/AddForm.js]: - [ ] Kindly avoid disabling linters as they help us write clean code. Remove these disabling lines and fix the error.

File: src/components/Book.js [hunk 3]
```diff
@@ -1,14 +1,18 @@
...
+import { useDispatch } from 'react-redux';
+import { deleteBook } from '../redux/books/booksSlice';
 
 // eslint-disable-next-line react/prop-types
 const Book = ({ book }) => {
-  // eslint-disable-next-line react/prop-types
-  const { title, author } = book;
+  const dispatch = useDispatch();
+  // eslint-disable-next-line react/prop-types, camelcase
+  const { item_id, title, author } = book;
```
Comment (TedLivist) [inline src/components/Book.js]: - [ ] Kindly avoid disabling linters as they help us write clean code. Remove this disabling line and fix the error.

- [ ] As for the prop type validation error you need to provide validations for your passed in props. You can visit [this tutorial](https://www.freecodecamp.org/news/how-to-use-proptypes-in-react/) on how to use prop types in react

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @Yacoubou-seidou,

Good job so far!
✔️ Use of local state for form component
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by AlphaNtihinduka — APPROVED** [review]
# STATUS: APPROVED :green_circle: :clap: 
## Hi @Yacoubou-seidou 🙋‍♂️, 

Your project is complete!  There is nothing else to say other than... Great job 👏 

approved-accepted

Congratulations! 🎉 
TO HIGHLIGHT:

- Your project is professional :heavy_check_mark: 
- Your repo is professional :heavy_check_mark: 
- Well-structured files :heavy_check_mark: 
- Professional README.md file 💯
- No linter errors :heavy_check_mark: 

Now you can merge this branch and move on :1st_place_medal: 


## Optional suggestions
- Kindly refer to the inline optional suggestions made. Ensure to apply them to have a professional project that suits recruiters' needs.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

HAPPY CODING AND GOOD LUCK WITH THE COMING ONES :crossed_fingers: :four_leaf_clover:

**Pull request merged**